tsl/src/compression/compression.c

```diff
@@ -1761,11 +1623,34 @@ row_decompressor_decompress_row(RowDecompressor *decompressor)
 			HeapTuple decompressed_tuple = heap_form_tuple(decompressor->out_desc,
 														   decompressor->decompressed_datums,
 														   decompressor->decompressed_is_nulls);
-			heap_insert(decompressor->out_rel,
-						decompressed_tuple,
-						decompressor->mycid,
-						0 /*=options*/,
-						decompressor->bistate);
+			if (tuplestorestate == NULL)
+			{
+				heap_insert(decompressor->out_rel,
+							decompressed_tuple,
+							decompressor->mycid,
+							0 /*=options*/,
+							decompressor->bistate);
+			}
+			else
+			{
+				TupleTableSlot *slot =
+					MakeSingleTupleTableSlot(decompressor->out_desc, &TTSOpsVirtual);
+
+				// create the virtual tuple slot
+				ExecClearTuple(slot); // might not need this
```

Comment:
Please check this in the final version of the PR.

Reply:
sorry for the confusion here. I believe it is needed, so I removed the misleading comment. From `ExecStoreVirtualTuple`: 
```
 * The protocol for loading a slot with virtual tuple data is:
 *		* Call ExecClearTuple to mark the slot empty.
 *		* Store data into the Datum/isnull arrays.
 *		* Call ExecStoreVirtualTuple to mark the slot valid.
 ```

Reply:
👍  It seems it is needed here. So, could you remove the comment?